plugins/libplugin-pay.c

```diff
@@ -2266,19 +2268,27 @@ void payment_continue(struct payment *p)
 {
 	struct payment_modifier *mod;
 	void *moddata;
+
+	trace_span_start("payment_continue", p);
 	/* If we are in the middle of calling the modifiers, continue calling
 	 * them, otherwise we can continue with the payment state-machine. */
 	p->current_modifier++;
 	mod = p->modifiers[p->current_modifier];
 
 	if (mod != NULL) {
+		char str[2];
```

Comment:
We generally don't allow the use of `sprintf` as it isn't safe (it takes a buffer it doesn't know the size of). Please switch this to `snprintf` which errors out if the buffer is too small for the format. Or better `tal_fmt(tmpctx, "%d", num);` will do all of it at once, including fixing the buffer overflow should we ever have 100+ modifiers 😋

Reply:
I saw you already corrected that in #7575.
Should I fix that in this commit? Or close this PR and go with your PR there?

Reply:
However we want to merge them. We can merge #7575 and your commit should still be credited to you 🤗 Or we fix up this one, merge this and then my follow-up PR.

By the way, if you check the allow maintainer checkbox when opening a PR you allow us to amend, fix up and fine tune PRs in situ 😉

Reply:
Fixed it up in this one. Although this one includes your commit now :-S